[Bug] Can't share to AnkiDroid if an image occlusion note type is selected

###### Reproduction Steps

1. Select an image occlusion note type
2. Share an image OR share text to AnkiDroid

Image fails with:

```
FATAL EXCEPTION: main
Process: com.ichi2.anki.debug, PID: 20368
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1582825430, result=-1, data=Intent { (has extras) }} to activity {com.ichi2.anki.debug/com.ichi2.anki.NoteEditor}: java.lang.NullPointerException
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5323)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7872)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.NullPointerException
	at com.ichi2.anki.NoteEditor$requestMultiMediaEditLauncher$1.invoke(NoteEditor.kt:209)
	at com.ichi2.anki.NoteEditor$requestMultiMediaEditLauncher$1.invoke(NoteEditor.kt:194)
	at com.ichi2.anki.NoteEditor$NoteEditorActivityResultCallback.onActivityResult(NoteEditor.kt:284)
	at com.ichi2.anki.NoteEditor$NoteEditorActivityResultCallback.onActivityResult(NoteEditor.kt:278)
	at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:414)
	at androidx.activity.result.ActivityResultRegistry.dispatchResult(ActivityResultRegistry.java:371)
	at androidx.activity.ComponentActivity.onActivityResult(ComponentActivity.java:845)
	at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:151)
	at android.app.Activity.dispatchActivityResult(Activity.java:8628)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5316)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362) 
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67) 
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7872) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 

```

Text fails with:

```
05:01:31.970 ACRA                                 E  ACRA caught a NullPointerException for com.ichi2.anki.debug
                                                     java.lang.NullPointerException
                                                     	at com.ichi2.anki.NoteEditor.onCollectionLoaded(NoteEditor.kt:608)
                                                     	at com.ichi2.anki.AnkiActivity.startLoadingCollection$lambda$0(AnkiActivity.kt:287)
                                                     	at com.ichi2.anki.AnkiActivity.$r8$lambda$kj9hf0g17vxBfzngPYSgjzadw2g(Unknown Source:0)
                                                     	at com.ichi2.anki.AnkiActivity$$ExternalSyntheticLambda1.execute(Unknown Source:2)
                                                     	at com.ichi2.async.CollectionLoader$load$1.invokeSuspend(CollectionLoader.kt:52)
                                                     	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                     	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                     	at android.os.Handler.handleCallback(Handler.java:942)
                                                     	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                     	at android.os.Looper.loopOnce(Looper.java:201)
                                                     	at android.os.Looper.loop(Looper.java:288)
                                                     	at android.app.ActivityThread.main(ActivityThread.java:7872)
                                                     	at java.lang.reflect.Method.invoke(Native Method)
                                                     	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                     	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
                                                     	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@232815d, Dispatchers.Main.immediate]
```

###### Research
AnkiDroid Version = 2.17beta5-debug (89ab6369ac90daf348ac7f5e4338a5ca9880a11c)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 13 (SDK 33)

ProductFlavor = amazon

Manufacturer = Google

Model = sdk_gphone64_arm64

Hardware = ranchu

Webview User Agent = Mozilla/5.0 (Linux; Android 13; sdk_gphone64_arm64 Build/TE1A.220922.034; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/103.0.5060.71 Mobile Safari/537.36

ACRA UUID = bdcca282-35c1-4097-9b3a-5936f403928b

Crash Reports Enabled = false

--- a/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt
@@ -98,6 +98,7 @@ import com.ichi2.libanki.Notetypes.Companion.NOT_FOUND_NOTE_TYPE
 import com.ichi2.utils.*
 import com.ichi2.utils.IntentUtil.resolveMimeType
 import com.ichi2.widget.WidgetStatus
+import kotlinx.coroutines.launch
 import org.json.JSONArray
 import org.json.JSONObject
 import timber.log.Timber
@@ -592,6 +593,23 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
             // set information transferred by intent
             var contents: String? = null
             val tags = intent.getStringArrayExtra(EXTRA_TAGS)
+
+            try {
+                // If content has been shared, we can't share to an image occlusion note type
+                if (currentNotetypeIsImageOcclusion() && (sourceText != null || caller == CALLER_ADD_IMAGE)) {
+                    val model = col.notetypes.all().first {
+                        !it.isImageOcclusion
+                    }
+                    changeNoteType(model.id)
+                }
+            } catch (e: NoSuchElementException) {
+                showSnackbar(R.string.missing_note_type)
+                // setting the text to null & caller to CALLER_NO_CALLER would skip adding text/image to edit field
+                sourceText = null
+                caller = CALLER_NO_CALLER
+                Timber.w(e)
+            }
+
             if (sourceText != null) {
                 if (aedictIntent && mEditFields!!.size == 3 && sourceText!![1]!!.contains("[")) {
                     contents = sourceText!![1]!!
@@ -2101,6 +2119,35 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
         requestIOEditorCloser.launch(intent)
     }
 
+    private fun changeNoteType(newId: NoteTypeId) {
+        val oldModelId = getColUnsafe.notetypes.current().getLong("id")
+        Timber.i("Changing note type to '%d", newId)
+
+        if (oldModelId == newId) {
+            return
+        }
+
+        val model = getColUnsafe.notetypes.get(newId)
+        if (model == null) {
+            Timber.w("New model %s not found, not changing note type", newId)
+            return
+        }
+
+        getColUnsafe.notetypes.setCurrent(model)
+        val currentDeck = getColUnsafe.decks.current()
+        currentDeck.put("mid", newId)
+        getColUnsafe.decks.save(currentDeck)
+
+        // Update deck
+        if (!getColUnsafe.config.getBool(ConfigKey.Bool.ADDING_DEFAULTS_TO_CURRENT_DECK)) {
+            deckId = model.optLong("did", Consts.DEFAULT_DECK_ID)
+        }
+
+        refreshNoteData(FieldChangeType.changeFieldCount(shouldReplaceNewlines()))
+        setDuplicateFieldStyles()
+        mDeckSpinnerSelection!!.updateDeckPosition(deckId)
+    }
+
     // ----------------------------------------------------------------------------
     // INNER CLASSES
     // ----------------------------------------------------------------------------
@@ -2110,27 +2157,7 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
             // Timber.i("NoteEditor:: onItemSelected() fired on mNoteTypeSpinner");
             // In case the type is changed while adding the card, the menu options need to be invalidated
             invalidateMenu()
-            val oldModelId = getColUnsafe.notetypes.current().getLong("id")
-            val newId = mAllModelIds!![pos]
-            Timber.i("Changing note type to '%d", newId)
-            if (oldModelId != newId) {
-                val model = getColUnsafe.notetypes.get(newId)
-                if (model == null) {
-                    Timber.w("New model %s not found, not changing note type", newId)
-                    return
-                }
-                getColUnsafe.notetypes.setCurrent(model)
-                val currentDeck = getColUnsafe.decks.current()
-                currentDeck.put("mid", newId)
-                getColUnsafe.decks.save(currentDeck)
-                // Update deck
-                if (!getColUnsafe.config.getBool(ConfigKey.Bool.ADDING_DEFAULTS_TO_CURRENT_DECK)) {
-                    deckId = model.optLong("did", Consts.DEFAULT_DECK_ID)
-                }
-                refreshNoteData(FieldChangeType.changeFieldCount(shouldReplaceNewlines()))
-                setDuplicateFieldStyles()
-                mDeckSpinnerSelection!!.updateDeckPosition(deckId)
-            }
+            changeNoteType(mAllModelIds!![pos])
         }
 
         override fun onNothingSelected(parent: AdapterView<*>?) {

--- a/AnkiDroid/src/main/res/values/02-strings.xml
+++ b/AnkiDroid/src/main/res/values/02-strings.xml
@@ -107,6 +107,7 @@
     <string name="select">Select</string>
     <string name="field_remapping">%1$s (from “%2$s”)</string>
     <string name="confirm_map_cards_to_nothing">Any cards mapped to nothing will be deleted. If a note has no remaining cards, it will be lost. Are you sure you want to continue?</string>
+    <string name="missing_note_type">No note type found</string>
 
     <!-- Timebox -->
     <string name="timebox_reached_title">Timebox reached</string>

